### PR TITLE
GCW-2906-User can specify sorting of offers

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-2906

### What does this PR do?

FEATURE: Adding Sorting feature on Offers list page

### Impacted Areas

Offers list/search page

### Mockup Link

https://xd.adobe.com/view/224f7314-f1c0-40b6-7d7a-929e616be6bb-547e/screen/d9e55a7a-570f-4805-9224-93be92764702/iPhone-6-7-8-105?x_product=xd-jiraserver%2F1.0

### CODE REVIEW

File: app/constants/offer-sorting-options.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export const offerSortingOptions = [
```
Comment (patrixr) [inline app/constants/offer-sorting-options.js]: Nice job extracting a config, makes me wonder whether the `constants` would be better named as `config` or something like that

File: app/constants/offer-sorting-options.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  {
+    column_name: "schedules.scheduled_at",
+    column_alias: "Due date",
```
Comment (patrixr) [inline app/constants/offer-sorting-options.js]: We probably want to have that translated, you can maybe input the translation key here

File: app/services/filter-service.js [hunk 3]
```diff
@@ -52,6 +52,22 @@ export default Ember.Service.extend(Ember.Evented, {
     this.set("offerStateFilters", []);
   },
 
+  offerSorting: PERSISTENT_VAR("offerSorting", null),
```
Comment (patrixr) [inline app/services/filter-service.js]:
```suggestion
 offerSorting: PERSISTENT_VAR("offerSorting", {}),
```
(default value)

File: app/services/filter-service.js [hunk 4]
```diff
@@ -52,6 +52,22 @@ export default Ember.Service.extend(Ember.Evented, {
...
+
+  init() {
+    this._super(...arguments);
+    this.set("openOfferSortOption", false);
+  },
+
+  applySortOn(column, is_desc = false) {
+    this.set("offerSorting", { column, is_desc });
+    this.toggleSortPage();
```
Comment (patrixr) [inline app/services/filter-service.js]: if `applySortOn` is called while the "sort page" is closed, the `toggle` will open the page.
So you probably want a `set('openOfferSortOption', false)` instead

File: app/templates/_global_components.hbs [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{{!-- Service-backed components --}}
+
+{{offer-sort-overlay
```
Comment (patrixr) [inline app/templates/_global_components.hbs]: I might be wrong, but I think this is only used in one place of the app (right ?)
So you could just add it to `search.hbs` without having it as an app level component.

File: app/services/filter-service.js (hunk 3 above)
Comment (abulsayyad123) [reply app/services/filter-service.js]: Yes, this is better :)

**Pull request merged**